in_toto/runlib.py

```diff
@@ -82,6 +92,14 @@ def record_artifacts_as_dict(artifacts):
     The files that result form a link command execution are called
     products.
 
+    Excludes files that are matched by the file patterns specified in
+    ARTIFACT_EXCLUDES setting.
+    Note: Exclude patterns are applied on the basename, i.e. the pattern
+    "foo" excludes /foo as well as /subdir/foo
```

Comment:
I see that this is a consequence of using fnmatch. Maybe in the future we want to change this behavior? (e.g,. have `**/.git` to ignore all .git folders in any subdirectory and `.git` to only ignore the one in root).